### PR TITLE
Adding Fax Machines

### PR DESCRIPTION
**Pull request by Alex-Selvey**
I have added the bulk of everything except for the map changes, but need help fixing the 13 errors that are popping up. Can't figure out what they need to be changed too. 

:cl: Paradise, Funce, AndrewMontagne, ported by MemeProof
add: Coding for faxes
add: Add fax rect
soundadd: added fax sound
imageadd: fax sprites (created by AndrewMontagne)

NEED:
mapedit: add sprites to map
/:cl:

### CODE REVIEW

File: code/modules/admin/admin_verbs.dm [hunk 1]
```diff
@@ -104,7 +104,8 @@ GLOBAL_LIST_INIT(admin_verbs_fun, list(
 	/client/proc/polymorph_all,
 	/client/proc/show_tip,
 	/client/proc/smite,
-	/client/proc/admin_away
+	/client/proc/admin_away,
+	/client/proc/fax_panel // don't know how to override this
```
Comment (ghost) [inline code/modules/admin/admin_verbs.dm]:
```suggestion
	/client/proc/fax_panel // sunset -- add fax panel
```

File: code/modules/admin/admin_verbs.dm [hunk 2]
```diff
@@ -238,7 +239,8 @@ GLOBAL_LIST_INIT(admin_verbs_hideable, list(
 	/client/proc/toggle_nuke,
 	/client/proc/cmd_display_del_log,
 	/client/proc/toggle_combo_hud,
-	/client/proc/debug_huds
+	/client/proc/debug_huds,
+	/client/proc/fax_panel //don't know how to override this
```
Comment (ghost) [inline code/modules/admin/admin_verbs.dm]:
```suggestion
	/client/proc/fax_panel // sunset -- add fax panel
```

File: sunset/code/__HELPERS/unique_ids.dm [hunk 3]
```diff
@@ -0,0 +1,44 @@
+// Unique Datum Identifiers
```
Comment (ghost) [inline sunset/code/__HELPERS/unique_ids.dm:1]: Da fuck is this?

File: sunset/code/modules/admin/admin.dm [hunk 4]
```diff
@@ -1,3 +1,5 @@
+#define ADMIN_FAX(user, fax, faxtype, sent) "(<a href='?_src_=holder;[HrefToken(TRUE)];AdminFaxCreate=[REF(user)];originfax=[REF(fax)];faxtype=[faxtype];replyto=[REF(sent)]'>FAX</a>)"
```
Comment (ghost) [inline sunset/code/modules/admin/admin.dm]: Stick it in ~sunset_defines/admin.dm

File: sunset/code/modules/admin/admin_verbs.dm [hunk 5]
```diff
@@ -0,0 +1,93 @@
+/* Need to figure out how to override this
```
Comment (ghost) [inline sunset/code/modules/admin/admin_verbs.dm]: You can't.

File: sunset/code/modules/paperwork/faxmachine.dm [hunk 6]
```diff
@@ -0,0 +1,329 @@
+GLOBAL_LIST_EMPTY(allfaxes)
+GLOBAL_LIST_INIT(admin_departments, list("Central Command"))
+GLOBAL_LIST_INIT(hidden_admin_departments, list("Syndicate"))
+GLOBAL_LIST_EMPTY(alldepartments)
+
+/obj/machinery/photocopier/faxmachine
```
Comment (ghost) [inline sunset/code/modules/paperwork/faxmachine.dm:6]: Does this NEED to be a photocopier subtype?

File: sunset/code/modules/paperwork/faxmachine.dm [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+
+/obj/machinery/photocopier/faxmachine/attackby(obj/item/item, mob/user, params)
+	if(istype(item,/obj/item/card/id) && !scan)
+		scan(item)
+	else if(istype(item, /obj/item/paper)) // Only paper can go in this one
+		return ..()
+	else
+		return
+
+/obj/machinery/photocopier/faxmachine/emag_act(mob/user)
```
Comment (ghost) [inline sunset/code/modules/paperwork/faxmachine.dm:62]:
```suggestion
/obj/machinery/photocopier/faxmachine/emag_act(mob/user)
	if(!(obj_flags & EMAGGED)
	  	obj_flags |= EMAGGED
  		to_chat(user, "<span class='notice'>The transmitters realign to an unknown source!</span>")
 	else
  		to_chat(user, "<span class='warning'>You swipe the card through [src], but nothing happens.</span>")
```

File: tgui/src/interfaces/faxmachine.ract [hunk 8]
```diff
@@ -0,0 +1,30 @@
+<ui-display title='Authorization'>
```
Comment (ghost) [inline tgui/src/interfaces/faxmachine.ract:1]: Recompile tgui!

**Review by ghost — CHANGES_REQUESTED** [review]

File: sunset/code/__HELPERS/unique_ids.dm (hunk 3 above)
Comment (Alex-Selvey) [reply sunset/code/__HELPERS/unique_ids.dm:1]: It is apart of faxes and if removed causes issues. However, once the other problems can be corrected, I'll see about removing this.

File: tgui/src/interfaces/faxmachine.ract (hunk 8 above)
Comment (Alex-Selvey) [reply tgui/src/interfaces/faxmachine.ract:1]: I am! I just need to sort out the other issues so i can at least get it to compile. It's on my todo list.

Comment (Alex-Selvey) [reply tgui/src/interfaces/faxmachine.ract:1]: Compiled

File: sunset/code/modules/admin/topic.dm [hunk 9]
```diff
@@ -9,4 +9,204 @@
...
+		P.y = rand(-1, 2)
+
+		if(destination != "All Departments")
+			if(fax.receivefax(P) == FALSE)
+				to_chat(owner, "<span class='warning'>Message transmission failed.</span>")
+				return
+		else
+			for(var/thing in GLOB.allfaxes)
+				var/obj/machinery/photocopier/faxmachine/F = thing
+				if(F.z in GLOB.station_z_levels)
```
[the remark this answers is not in the document]
Comment (Alex-Selvey) [reply sunset/code/modules/admin/topic.dm]: Fixed

File: code/modules/admin/admin_verbs.dm (hunk 1 above)
Comment (Alex-Selvey) [reply code/modules/admin/admin_verbs.dm]: Fixed

File: code/modules/admin/admin_verbs.dm (hunk 2 above)
Comment (Alex-Selvey) [reply code/modules/admin/admin_verbs.dm]: Fixed

File: sunset/code/modules/admin/admin.dm (hunk 4 above)
Comment (Alex-Selvey) [reply sunset/code/modules/admin/admin.dm]: Fixed

File: sunset/code/modules/admin/admin_verbs.dm (hunk 5 above)
Comment (Alex-Selvey) [reply sunset/code/modules/admin/admin_verbs.dm]: Removed file

File: sunset/code/modules/paperwork/faxmachine.dm (hunk 7 above)
Comment (Alex-Selvey) [reply sunset/code/modules/paperwork/faxmachine.dm:62]: Fixed

**Comment by SunsetStationuser** [comment]
@MemeProof  any update on this?

**Comment by Alex-Selvey** [comment]
Needs to be rechecked, but I don't know how to setup faxes. I'll cue this up for testing this weekend.

**Comment by Alex-Selvey** [comment]
Completely threw away old branch and redid the port. Creating new PR.

**Pull request closed without merging**